src/layout/Menu/Menu.style.ts

```diff
@@ -24,21 +24,34 @@ export const Action = styled.button`
   }
 `;
 
-export const Toggle = styled.div<{ open?: boolean }>`
-  width: 1.5rem;
-  height: 1.5rem;
+export const Toggle = styled.button<{
+  open?: boolean;
+  indentation: number;
+}>`
+  width: 1.25rem;
+  height: 1.25rem;
   position: absolute;
-  top: 0.5rem;
-  left: -0.5rem;
-  transition: 120ms ease-in-out;
-
-  transform: ${(p) => (p.open ? 'rotate(0deg)' : 'rotate(-90deg)')};
+  top: 0.625rem;
+  left: ${({ indentation }) => rem(indentation)};
 
   &:focus {
     outline: none;
   }
 
+  &:hover {
+    background: ${({ theme }) =>
+      theme.name === 'dark' ? grayscale(600) : slate(200)};
+  }
```

Comment:
There are multiple `theme` ternaries throughout the style code. Can all or some of them be replaced by tokens from `core`? For this that can't, they should be named constants defined outside the styled blocks and reused.

Reply:
Asking design if there are tokens we can use. If not, since the colors are based on theme, should I just create a function that will read the theme and return the appropriate color?